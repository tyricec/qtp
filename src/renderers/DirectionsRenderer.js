export default class DirectionsRenderer {
  constructor(element, directions) {
    this.container = element
    update(directions).call(this)
  }

}

function update() {
  this.list = {

  }
}