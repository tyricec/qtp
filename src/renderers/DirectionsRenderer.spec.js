import DirectionsRenderer from './DirectionsRenderer'

test('DirectionsRenderer renders directions where specified', () => {
  const directions = [{
    maneuver: 'turn-left',
    instruction: 'Turn Left',
  }, {
    maneuver: 'merge-right',
    instruction: 'Merge Right',
  }, {
    maneuver: 'straight',
    instruction: 'Head straight and destination is to the left',
  },]

  const element = document.createElement('div')

  const directionsRenderer = new DirectionsRenderer(element, directions)

  directionsRenderer.render()

  expect(element.outerHTML).toMatchSnapshot()
})