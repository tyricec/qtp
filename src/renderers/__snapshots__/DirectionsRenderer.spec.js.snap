// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionsRenderer renders directions where specified 1`] = `"<div><ul class=\\"qtp-results__list\\"><li class=\\"qtp-results__list-item\\"><span class=\\"icon-turn-left qtp-results__icon\\"></span><div>Turn Left</div><hr class=\\"qtp-results__list-hr\\"></li><li class=\\"qtp-results__list-item\\"><div>Merge Right</div><hr class=\\"qtp-results__list-hr\\"></li><li class=\\"qtp-results__list-item\\"><span class=\\"icon-straight qtp-results__icon\\"></span><div>Head straight and destination is to the left</div><hr class=\\"qtp-results__list-hr\\"></li></ul></div>"`;
